src/ion.hpp

```diff
@@ -71,19 +74,50 @@ public :
                +  sizeof(ion);
     }
 
+    void set_valency(int v) {
+        valency_ = v;
+    }
+    int valency() const {
+        return valency_;
+    }
+
+    void set_default_internal_concentration(value_type c) {
+        EXPECTS(c>=value_type(0));
+        Xi_default_ = c;
+    }
+    void set_default_external_concentration(value_type c) {
+        EXPECTS(c>=value_type(0));
+        Xo_default_ = c;
+    }
+
```

Comment:
Why not just expose the member variables for valency and default concentrations (after a suitable renaming)? This would be my preference as it makes the code smaller and the effect transparent.

Reply:
done